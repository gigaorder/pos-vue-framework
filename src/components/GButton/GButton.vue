<template>
	<button :class="buttonClass" :style="buttonStyle" @click="onClick">
		<slot>{{text}}</slot>
	</button>
</template>

<script>
  export default {
    name: 'GButton',
    props: {
      raised: Boolean,
      depressed: Boolean,
      disabled: Boolean,
      flat: Boolean,
      icon: Boolean,
      outline: Boolean,
      dashed: Boolean,
      width: String,
      height: String,
      color: String,
      textColor: String,
      text: String
    },
    computed: {
      // susceptible to change per usage -> use composition/HOC
      buttonClass() {
        return {
          'btn-raised': this.raised,
          'btn-depressed': this.depressed,
          'disabled': this.disabled,
          'btn-icon': this.icon,
          'btn-outline': this.outline,
          'b-dashed': this.dashed
        }
      },
      buttonStyle() {
        let style = {};
        if (this.height) {
          Object.assign(style, { height: this.height });
        }
        if (this.width) {
          Object.assign(style, { width: this.width });
        }
        if (this.outline) {
          Object.assign(style, {
            'border-color': this.color,
            color: this.color
          });
        } else if (this.color) {
          Object.assign(style, { 'border-color': 'transparent', 'background-color': this.color });
        }
        if (this.flat) {
          Object.assign(style, {
            'border-color': 'transparent',
            'background-color': 'transparent'
          });
        }
        if (this.textColor) {
          Object.assign(style, { color: this.textColor });
        }
        return style;
      }
    },
    methods: {
      onClick(event) {
        this.$emit('click', event);
      }
    }
  }
</script>

<style scoped>

</style>