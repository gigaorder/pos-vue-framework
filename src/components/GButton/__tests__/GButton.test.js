import GButton from '../GButton';
import { mount } from '@vue/test-utils';

describe('Button', () => {
  let mountFunction

  beforeEach(() => {
    mountFunction = (options = {}) => {
      return mount(GButton, {
        ...options
      })
    }
  })

  // match selector
  it('has a button', () => {
    const wrapper = mountFunction()
    expect(wrapper.contains('button')).toBe(true)
  })

  // match snapshot
  it('should have raised class', () => {
    const wrapper = mountFunction({
      propsData: {
        raised: true
      }
    })
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should emit click event', () => {
    const wrapper = mountFunction()
    wrapper.trigger('click')
    expect(wrapper.emitted()).toBeTruthy()
  });

  it('should have correct bgStyle', function () {
    const wrapper = mountFunction({
      propsData: {
        height: 40,
        width: 100,
        color: 'blue'
      }
    })
    expect(wrapper.html()).toMatchSnapshot()
  });
})
