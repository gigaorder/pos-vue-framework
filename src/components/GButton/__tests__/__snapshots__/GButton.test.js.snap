// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should have correct style 1`] = `"<button class=\\"\\" style=\\"border-color: transparent; background-color: blue;\\"></button>"`;

exports[`Button should have raised class 1`] = `"<button class=\\"btn-raised\\"></button>"`;
